rts/Rendering/UniformConstants.cpp

```diff
@@ -0,0 +1,132 @@
+#include "UniformConstants.h"
+
+#include <cassert>
+
+#include "Rendering/GlobalRendering.h"
+#include "Rendering/ShadowHandler.h"
+#include "Game/Camera.h"
+#include "Game/CameraHandler.h"
+#include "Game/GlobalUnsynced.h"
+#include "Sim/Misc/GlobalConstants.h"
+#include "Sim/Misc/GlobalSynced.h"
+#include "Map/ReadMap.h"
+#include "System/Log/ILog.h"
+#include "System/SafeUtil.h"
+
+
+void UniformConstants::Init()
+{
+	if (!Supported()) {
+		LOG_L(L_ERROR, "[UniformConstants::%s] Important OpenGL extensions are not supported by the system\n  GLEW_ARB_uniform_buffer_object = %d\n  GLEW_ARB_shading_language_420pack = %d", __func__, GLEW_ARB_uniform_buffer_object, GLEW_ARB_shading_language_420pack);
+		return;
+	}
+
+	{
+		GLint uniformBufferOffset = 0;
+
+		glGetIntegerv(GL_UNIFORM_BUFFER_OFFSET_ALIGNMENT, &uniformBufferOffset);
+
+		umbBufferSize = ((sizeof(UniformMatricesBuffer) / uniformBufferOffset) + 1) * uniformBufferOffset;
+		upbBufferSize = ((sizeof(UniformParamsBuffer) / uniformBufferOffset) + 1) * uniformBufferOffset;
+	}
+
+	for (auto& buf : umbBuffers) {
```

Comment:
as usual, avoid auto

Reply:
This led to the following fugliness: `for (VBO*& buf : umbBuffers) {`